matching-notification-migrations/src/main/java/org/phenotips/matchingnotification/internal/R74696PatientNetwork456DataMigration.java

```diff
@@ -0,0 +1,149 @@
+ *
+ * @version $Id$
+ * @since 1.3m1
+ */
+@Component
+@Named("R74696PatientNetwork456")
+@Singleton
+public class R74696PatientNetwork456DataMigration extends AbstractHibernateDataMigration
+{
+    // MariaDB/MySQL do not allow to update a table that is used in a sub-query
+    // ("ERROR 1093 (HY000): You can't specify target table '...' for update in FROM clause").
+    // So an additional sub-sub-query is used to avoid using two separate "SELECT" and "DELETE" queries
+    // https://stackoverflow.com/questions/45494/mysql-error-1093-cant-specify-target-table-for-update-in-from-clause
+    private static final String SQL_DELETE_NON_NOTIFIED_COPIES_OF_A_NOTIFIED_MATCH =
+        "delete from patient_matching d where d.notified = false and exists"
+        + " (select * from (select * from patient_matching) as d2 where"
+        + " d2.referencePatientId = d.referencePatientId and d2.matchedPatientId = d.matchedPatientId"
+        + " and d2.matchedServerId = d.matchedServerId and d2.referenceServerId = d.referenceServerId"
+        + " and d2.notified = true)";
+
+    private static final String SQL_DELETE_NON_REJECTED_COPIES_OF_A_REJECTED_MATCH =
+        "delete from patient_matching d where d.status = 'uncategorized' and exists"
+        + " (select * from (select * from patient_matching) as d2 where"
+        + " d2.referencePatientId = d.referencePatientId and d2.matchedPatientId = d.matchedPatientId"
+        + " and d2.matchedServerId = d.matchedServerId and d2.referenceServerId = d.referenceServerId"
+        + " and d2.status = 'rejected')";
+
+    private static final String SQL_DELETE_INVERSE_COPIES_OF_LOCAL_MATCHES =
+        "delete from patient_matching d where d.matchedServerId = '' and d.referenceServerId = ''"
+        + " and d.referencePatientId > d.matchedPatientId and exists"
+        + " (select * from (select * from patient_matching) as d2 where"
+        + " d2.referencePatientId = d.matchedPatientId and d2.matchedPatientId = d.referencePatientId"
+        + " and d2.matchedServerId = '' and d2.referenceServerId = '')";
+
+    private static final String SQL_NORMALIZE_ORDER_OF_PATIENTS_IN_LOCAL_MATCHES =
+        "update patient_matching set referencePatientId = matchedPatientId, matchedPatientId = referencePatientId"
+        + ", referenceDetails = matchedDetails, matchedDetails = referenceDetails"
+        + " where referencePatientId > matchedPatientId and referenceServerId = '' and matchedServerId = ''";
+
+    /** Logging helper object. */
+    @Inject
+    private Logger logger;
+
+    /** Handles persistence. */
+    @Inject
+    private HibernateSessionFactory sessionFactory;
+
+    @Override
+    public String getDescription()
+    {
+        return "Remove duplicate matches and normalize order of patients in a local match";
+    }
+
+    @Override
+    public XWikiDBVersion getVersion()
+    {
+        return new XWikiDBVersion(74697);
```

Comment:
This does not match the version in the classname.

Reply:
Right, committed a debug DB version by accident